app/user.go

```diff
@@ -0,0 +1,597 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/temporalio/tcld/protogen/api/auth/v1"
+	"github.com/temporalio/tcld/protogen/api/authservice/v1"
+	"github.com/urfave/cli/v2"
+	"strings"
+)
+
+const (
+	userIDFlagName              = "user-id"
+	userEmailFlagName           = "user-email"
+	accountRoleFlagName         = "account-role"
+	namespacePermissionFlagName = "namespace-permission"
+)
+
+var (
+	userIDFlag = &cli.StringFlag{
+		Name:    userIDFlagName,
+		Usage:   "The user id",
+		Aliases: []string{"i"},
+	}
+	userEmailFlag = &cli.StringFlag{
+		Name:    userEmailFlagName,
+		Usage:   "The user email address of the user to be invited",
+		Aliases: []string{"e"},
+	}
+)
+
+type UserClient struct {
+	client authservice.AuthServiceClient
+	ctx    context.Context
+}
+
+type GetUserClientFn func(ctx *cli.Context) (*UserClient, error)
+
+func GetUserClient(ctx *cli.Context) (*UserClient, error) {
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &UserClient{
+		client: authservice.NewAuthServiceClient(conn),
+		ctx:    ct,
+	}, nil
+}
+
+func (c *UserClient) listUsers(
+	namespace string,
+) error {
+	totalRes := &authservice.GetUsersResponse{}
+	pageToken := ""
+	for {
+		res, err := c.client.GetUsers(c.ctx, &authservice.GetUsersRequest{
+			PageToken: pageToken,
+			Namespace: namespace,
+		})
+		if err != nil {
+			return err
+		}
+		totalRes.Users = append(totalRes.Users, res.Users...)
+		// Check if we should continue paging
+		pageToken = res.NextPageToken
+		if len(pageToken) == 0 {
+			return PrintProto(totalRes)
+		}
+	}
+}
+
+func (c *UserClient) getUser(userID, userEmail string) (*auth.User, error) {
+	if userID == "" && userEmail == "" {
+		return nil, fmt.Errorf("both user-id and user-email not set")
+	}
+
+	if userID != "" && userEmail != "" {
```

Comment:
this is pretty neat, but we are currently using v2, and it seems the v3 package is alpha. Does this exist in v2?

Reply:
Doh you are right. Honestly we have been finding some issues with urfave (if you have an env variable which stores a secret it will print that secret in the help message)... but that is a separate discussion from this PR.